src/web/mjs/connectors/Pixiv.mjs

```diff
@@ -31,29 +41,37 @@ export default class Pixiv extends Connector {
 
     async _getChapters(manga) {
         let chapterList = [];
-        for (let page = 1, run = true; run; page++) {
-            const chapters = await this._getChaptersFromPage(manga.id, page);
-            chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+        if (manga.id.startsWith('artwork-')) {
+            chapterList.push({
+                id: manga.id.match(/artwork-(\d+)/)[1],
+                title: manga.title
+            });
+        } else {
+            for (let page = 1, run = true; run; page++) {
+                const chapters = await this._getChaptersFromPage(manga.id, page);
+                chapters.length > 0 ? chapterList.push(...chapters) : run = false;
+            }
         }
-        return chapterList.filter((chapter, index) => {
-            return index === chapterList.findIndex(item => chapter.id === item.id);
-        });
+        return chapterList;
     }
 
     async _getChaptersFromPage(mangaId, page) {
         const uri = new URL(`series/${mangaId}?p=${page}&lang=en`, this.apiURL);
         const request = new Request(uri, this.requestOptions);
         const data = await this.fetchJSON(request);
-        if (data.body.thumbnails.illust.length < 5)
-            return [];
-        return data.body.thumbnails.illust
-            .filter(chapter => chapter.seriesId === mangaId)
+        return data.body.page.series
             .map(chapter => {
-                return {
-                    id: chapter.id,
-                    title: chapter.title.trim()
-                };
-            });
+                const chapterContents = data.body.thumbnails.illust.find(c => c.id === chapter.workId);
+                if (chapterContents != undefined) {
+                    return {
+                        id: chapterContents.id,
+                        title: chapterContents.title.trim()
+                    };
+                } else {
+                    return null;
```

Comment:
Why not just mapping data.body.page.series to manually populate an array of chapter you can return later? 
I am not fond of letting null elements in chapter array but i may be wrong.
```

let chapterlist = []
const data = await this.fetchJSON(request);

data.body.page.series .map(chapter => {
    const chapterContents = data.body.thumbnails.illust.find(c => c.id === chapter.workId);
    if (chapterContents) {
         chapterlist .push(  { id: chapterContents.id,     title: chapterContents.title.trim()  });
      }// else do nothing 
}
return chapterlist 

```

Reply:
Changed the algorithm to populate the list in 12f9713 as you suggested, except I used `forEach` instead of `map`.